rust/cymbal/src/langs/js.rs

```diff
@@ -0,0 +1,70 @@
+use reqwest::Url;
+use serde::Deserialize;
+
+use crate::error::Error;
+
+// A minifed JS stack frame. Just the minimal information needed to lookup some
+// sourcemap for it and produce a "real" stack frame.
+// TODO - how do we know if this frame is minified? If it isn't, we can skip a lot of work, but I think we have to guess? Based on whether we can get a sourcemap for it?
```

Comment:
generally minified traces end with a sourceMappingUrl, if that doesn't exist it's either configured not to output it, or isn't minified 🙈 - but I think we can make this a strong requirement? No reason to ever not have the sourceMappingUrl, even if its relative and not publicly accessible..

Reply:
Just so I understand fully, you mean for a given frame in a stack trace, the URL ("filename") on that frame points to a file that has a `sourceMappingUrl` at the end - and that's the only way to know if that frame is minified?

And there's no way to know if a frame is minified or not _except_ making a http request to that associated file and looking for a source-map url?

Reply:
> and that's the only way to know if that frame is minified?

I don't think so, but _I think_ this is the only way to be sure. I see your point, would be nicer to be clever here somehow so we don't need to fetch random files to check this.